src/coreclr/ilasm/portable_pdb.cpp

```diff
@@ -225,7 +226,11 @@ HRESULT PortablePdbWriter::DefineSequencePoints(Method* method)
         // LocalSignature
         localSigRid = RidFromToken(method->m_LocalsSig);
         CompressUnsignedLong(localSigRid, blob);
-        // InitialDocument TODO: skip this for now
+        currDocumentRid = RidFromToken(method->m_FirstDocument->GetToken());
```

Comment:
Not in any realistic way.

This line is conditioned on method->m_LinePCList.COUNT() != 0 and the only way that can be true is the PUSH in https://github.com/dotnet/runtime/pull/102199/files/0a2f49bb7de5d1432b4d1bbcc5174ff26e0d910d#diff-1f5d27b028692df005d7f33a02b3080f52ffb3a4e1384b906a7bad7186f5e601R1444 which can't be reached without setting m_FirstDocument to instr->pOwnerDocument. This one in turn is set from the m_currentDocument set by PortablePdbWriter::DefineDocument called by Assembler::SetSourceFileName itself called directly from wmain with the starting file name. The m_currentDocument can only be reset to null by calling the PortablePdbWriter::Init method which is never called after constructing the object.

SetSourceFileName does nothing if the passed string is null or start with a zero character (empty string) but this cannot be the case for the invocation from wmain as the file name is forced by MakeProperSourceFileName to contain at least a dot (it would add ".il" to it if it didn't).

This leaves the very convoluted case : on a machine with signed characters, separate representation for +0 and -0 (i.e. not a two's complement machine) where the C++ ABI define only one of them as a string terminator (I didn't dig in the C++ standard to see if it would even be conformant) and where the other can be the first character of a file name, SetSourceFileName would ultimately leave a null value pass. I doubt this would be the only place that fails in such a corner case.